xbmc/guilib/GUIAudioManager.cpp

```diff
@@ -128,21 +134,19 @@ void CGUIAudioManager::PlayWindowSound(int id, WINDOW_SOUND event)
   if (!m_bEnabled)
     return;
 
-  windowSoundMap::iterator it=m_windowSoundMap.find(id);
+  const auto& it = m_windowSoundMap.find(id);
   if (it==m_windowSoundMap.end())
     return;
 
-  CWindowSounds sounds=it->second;
-  IAESound *sound = NULL;
-  switch (event)
-  {
-  case SOUND_INIT:
-    sound = sounds.initSound;
-    break;
-  case SOUND_DEINIT:
-    sound = sounds.deInitSound;
-    break;
-  }
+  const auto& sound = [&]() {
```

Comment:
does the lambda need to be bound to a ref, and what do the trailing `()` mean? sorry, never seen this before

Reply:
This is an immediately invoked lambda, so what's bound to the const ref is the result of executing that lambda and not the lambda itself.

By binding the result as a const ref I **thought** I can get around an unnecessary reference increment+decrement of the `std::shared_ptr,` but I have just realized that `std::shared_ptr` has a move constructor that makes this optimization unnecessary.

I'll change this to a regular `const auto` here and in the other places where it's unnecessary.

Reply:
aha, yeah now i got it :) then a const& is correct…. that auto totally confused me.
a good option maybe to explicitly `const std::shared_ptr<IAESound>& sound = …` in this case.
but let‘s just wait for further comments